### PR TITLE
Mecademic Arm Upgrades [Blocks]

### PR DESCRIPTION
**Pull request by TheBigSasha**
This PR Depends on #950 in Flojoy Studio for some changes to mecademic state handling.

This PR contains new nodes for moving and configuring Mecademic robot arms.

### CODE REVIEW

**Comment by itsjoeoui** [comment]
@TheBigSasha not sure if I understand the issue with log, but you can build locally with `pnpm build` which will give you the same log

**Review by jingxiangmo — CHANGES_REQUESTED** [review]
Remove .idea

**Review by jingxiangmo — DISMISSED** [review]

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/CALCULATE_CIRCLE_MOVE/CALCULATE_CIRCLE_MOVE.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+@safe_robot_operation
+@flojoy(deps={"mecademicpy": "1.4.0"})
+def CALCULATE_CIRCLE_MOVE(
+    center_X: Optional[float] = 0.0,
+    center_Y: Optional[float] = 0.0,
+    center_Z: Optional[float] = 0.0,
+    radius: Optional[float] = 0.0,
+    revolutions: Optional[float] = 1.0,
+    point_duration: Optional[int] = 500,
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/CALCULATE_CIRCLE_MOVE/CALCULATE_CIRCLE_MOVE.py]: There's no need to specify optional for all of these, having a default value is enough

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/CALCULATE_CIRCLE_MOVE/CALCULATE_CIRCLE_MOVE.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import pandas as pd
+from typing import Optional
+from flojoy import flojoy, DataFrame
+from PYTHON.utils.mecademic_state.mecademic_calculations import getCirclePositions
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/CALCULATE_CIRCLE_MOVE/CALCULATE_CIRCLE_MOVE.py]: I believe this name needs to be updated to snake_case

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/CALCULATE_CIRCLE_MOVE/CALCULATE_CIRCLE_MOVE.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    center_Y: Optional[float]
+        The Y coordinate of the center of the circle. If not specified, the default value of 0.0 is used.
+    center_Z: Optional[float]
+        The Z coordinate of the center of the circle. If not specified, the default value of 0.0 is used.
+    radius: Optional[float]
+        The radius of the circle. If not specified, the default value of 0.0 is used.
+    revolutions: Optional[float]
+        The number of revolutions to make. If not specified, the default value of 1.0 is used.
+    point_duration: Optional[int]
+        The duration of each point in milliseconds. If not specified, the default value of 500 is used.
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/CALCULATE_CIRCLE_MOVE/CALCULATE_CIRCLE_MOVE.py]:
```suggestion
    center_X : float
        The X coordinate of the center of the circle. If not specified, the default value of 0.0 is used.
    center_Y : float
        The Y coordinate of the center of the circle. If not specified, the default value of 0.0 is used.
    center_Z : float
        The Z coordinate of the center of the circle. If not specified, the default value of 0.0 is used.
    radius : float
        The radius of the circle. If not specified, the default value of 0.0 is used.
    revolutions : float
        The number of revolutions to make. If not specified, the default value of 1.0 is used.
    point_duration : int
        The duration of each point in milliseconds. If not specified, the default value of 500 is used.
```

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/CALCULATE_CIRCLE_MOVE/CALCULATE_CIRCLE_MOVE.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    Returns
+    -------
+    dataframe
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/CALCULATE_CIRCLE_MOVE/CALCULATE_CIRCLE_MOVE.py]:
```suggestion
    DataFrame
```

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/DELAY/DELAY.py [hunk 5]
```diff
@@ -1,26 +1,29 @@
...
     time: float,
 ) -> TextBlob:
     """
     Delay the action between two blocks.
-
     Parameters
     ----------
-    ip_address : TextBlob
+    Inputs
+    ------
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/DELAY/DELAY.py]:
```suggestion
    Parameters
    ----------
```

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/DISCONNECT/DISCONNECT.py [hunk 6]
```diff
@@ -1,15 +1,16 @@
...
 def DISCONNECT(ip_address: TextBlob) -> None:
     """
-    Disconnects the Mecademic robot arm via its API.
-
+    DISCONNECT disconnects the Mecademic robot arm via its API. This is required if the arm is to be used without rebooting.
     Parameters
     ----------
-    ip_address : TextBlob
+    Inputs
+    ------
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/DISCONNECT/DISCONNECT.py]:
```suggestion
    Parameters
    ----------
```

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/HOME/HOME.py [hunk 7]
```diff
@@ -1,20 +1,24 @@
...
 def HOME(ip_address: TextBlob) -> TextBlob:
     """
-    Home the robot arm. This block is required to be run before any other robot arm movement. It is recommended to run this block immediately after "ACTIVATE".
+    Home the robot arm. This block is required to be run before any other robot arm movement. It is recommended to run this block immediately after "ACTIVATE". The robot is expected to move a little bit during the running of this node.
 
     Parameters
     ----------
-    ip_address : TextBlob
+    Inputs
+    ------
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/HOME/HOME.py]:
```suggestion
    Parameters
    ----------
```

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_CIRCLE/MOVE_CIRCLE.py [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+from PYTHON.utils.mecademic_state.mecademic_state import query_for_handle
+from PYTHON.utils.mecademic_state.mecademic_helpers import safe_robot_operation
+
+
+@safe_robot_operation
+@flojoy(deps={"mecademicpy": "1.4.0"})
+def MOVE_CIRCLE(
+    ip_address: TextBlob,
+    radius: Optional[float] = 0.0,
+    revolutions: Optional[float] = 1.0,
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_CIRCLE/MOVE_CIRCLE.py]:
```suggestion
    radius: float = 0.0,
    revolutions: float = 1.0,
```

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_CIRCLE/MOVE_CIRCLE.py [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    ------
+    ip_address: TextBlob
+        The IP address of the robot arm.
+
+    Parameters
+    ----------
+    radius: Optional[float]
+        The radius of the circle. If not specified, the default value of 0.0 is used.
+    revolutions: Optional[float]
+        The number of revolutions to make. If not specified, the default value of 1.0 is used.
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_CIRCLE/MOVE_CIRCLE.py]:
```suggestion
    Parameters
    ----------
    ip_address: TextBlob
        The IP address of the robot arm.
    radius: float
        The radius of the circle. If not specified, the default value of 0.0 is used.
    revolutions: float
        The number of revolutions to make. If not specified, the default value of 1.0 is used.
```

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_CIRCLE/MOVE_CIRCLE.py [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+
+    Returns
+    -------
+    ip_address
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_CIRCLE/MOVE_CIRCLE.py]:
```suggestion
    TextBlob
```

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/HOME/HOME.py [hunk 11]
```diff
@@ -1,20 +1,24 @@
...
+    ip_address: TextBlob
         The IP address of the robot arm.
 
     Returns
     -------
-    ip_address : TextBlob
+    ip_address: TextBlob
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/HOME/HOME.py]:
```suggestion
    TextBlob
```

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/DELAY/DELAY.py [hunk 12]
```diff
@@ -1,26 +1,29 @@
...
         The IP address of the robot arm.
 
-    time : float
+    time: float
         The time of delay in seconds.
 
     Returns
     -------
-    ip_address : TextBlob
+    ip_address: TextBlob
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/DELAY/DELAY.py]:
```suggestion
    TextBlob
```

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/CONNECT/CONNECT.py [hunk 13]
```diff
@@ -1,22 +1,37 @@
...
     ----------
     ip_address : str
         The IP address of the robot arm.
+    simulator : bool
+        Whether to activate the simulator or not. Defaults to False.
+
 
     Returns
     -------
     String
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/CONNECT/CONNECT.py]:
```suggestion
    TextBlob
```

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_JOINT/MOVE_JOINT.py [hunk 14]
```diff
@@ -37,7 +40,7 @@ def MOVE_JOINT(
 
     Returns
     -------
-    ip_address : TextBlob
+    ip_address
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_JOINT/MOVE_JOINT.py]:
```suggestion
    TextBlob
```

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_KEYFRAMES/MOVE_KEYFRAMES.py [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+@safe_robot_operation
+@flojoy(deps={"mecademicpy": "1.4.0"})
+def MOVE_KEYFRAMES(ip_address: TextBlob, keyframes: DataFrame) -> TextBlob:
+    """
+    The MOVE_KEYFRAMES node linearly moves the robot's tool according to a set of 3d animation style keyframes.
+
+    Parameters
+    ----------
+    Inputs
+    ------
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_KEYFRAMES/MOVE_KEYFRAMES.py]:
```suggestion
```

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_KEYFRAMES/MOVE_KEYFRAMES.py [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+    ip_address: TextBlob
+        The IP address of the robot arm.
+    keyframes: DataFrame
+        A dataframe containing the keyframes to move to. The dataframe must have the following columns:
+        x, y, z, alpha, beta, gamma, duration. The duration column is the time in seconds to move to the next keyframe.
+
+    Returns
+    -------
+    ip_address
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_KEYFRAMES/MOVE_KEYFRAMES.py]:
```suggestion
    TextBlob
```

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_KEYFRAMES/MOVE_KEYFRAMES.py [hunk 17]
```diff
@@ -0,0 +1,53 @@
+from flojoy import flojoy, TextBlob, DataFrame
+from PYTHON.utils.mecademic_state.mecademic_calculations import calculateLimitingMaxVel
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_KEYFRAMES/MOVE_KEYFRAMES.py]: Seems like this should be updated to the snake_case name

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_KEYFRAMES_LIN/MOVE_KEYFRAMES_LIN.py [hunk 18]
```diff
@@ -0,0 +1,59 @@
...
+    """
+    The MOVE_KEYFRAMES node LINEARLY RELATIVE TO A REFERENCE FRAME moves the robot's tool according to a set of 3d animation style keyframes.
+
+    Set a tool reference frame (TRF) to move the robot's tool relative to the TRF. The TRF is set using the SET_TRF node.
+
+
+    Parameters
+    ----------
+    Inputs
+    ------
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_KEYFRAMES_LIN/MOVE_KEYFRAMES_LIN.py]:
```suggestion
```

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_KEYFRAMES_LIN/MOVE_KEYFRAMES_LIN.py [hunk 19]
```diff
@@ -0,0 +1,59 @@
+from flojoy import flojoy, TextBlob, DataFrame
+from PYTHON.utils.mecademic_state.mecademic_calculations import calculateLimitingMaxVel
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_KEYFRAMES_LIN/MOVE_KEYFRAMES_LIN.py]: update to snake_case

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_LIN/MOVE_LIN.py [hunk 20]
```diff
@@ -36,7 +40,7 @@ def MOVE_LIN(
 
     Returns
     -------
-    ip_address : TextBlob
+    ip_address: TextBlob
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_LIN/MOVE_LIN.py]:
```suggestion
    TextBlob
```

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_JOINT/MOVE_JOINT.py [hunk 21]
```diff
@@ -17,11 +18,13 @@ def MOVE_JOINT(
     """
     Linearly move the robot's tool to an absolute Cartesian position.
 
+    Inputs
+    ------
+    ip_address: TextBlob
+        The IP address of the robot arm.
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_JOINT/MOVE_JOINT.py]: We don't do the whole separate Inputs and Parameters sections thing anymore, you can just put it all under "Parameters"

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_LIN/MOVE_LIN.py [hunk 22]
```diff
@@ -17,10 +18,13 @@ def MOVE_LIN(
     """
     Linearly move the robot's tool to an absolute Cartesian position.
 
+    Inputs
+    ------
+    ip_address: TextBlob
+        The IP address of the robot arm.
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_LIN/MOVE_LIN.py]: Put under parameters

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_LIN_REL_TRF/MOVE_LIN_REL_TRF.py [hunk 23]
```diff
@@ -0,0 +1,50 @@
...
+@safe_robot_operation
+@flojoy(deps={"mecademicpy": "1.4.0"})
+def MOVE_LIN_REL_TRF(
+    ip_address: TextBlob,
+    x: float,
+    y: float,
+    z: float,
+    alpha: Optional[float] = 0,
+    beta: Optional[float] = 0,
+    gamma: Optional[float] = 0,
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_LIN_REL_TRF/MOVE_LIN_REL_TRF.py]:
```suggestion
    alpha: float = 0,
    beta: float = 0,
    gamma: float = 0,
```

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_LIN_REL_TRF/MOVE_LIN_REL_TRF.py [hunk 24]
```diff
@@ -0,0 +1,50 @@
...
+    y : float
+        The y coordinate of the position to move to
+    z : float
+        The z coordinate of the position to move to
+    alpha : float, optional
+        The alpha coordinate (rotation in radians about the x axis) of the position to move to.
+    beta : float, optional
+        The beta coordinate   (rotation in radians about the y axis) of the position to move to.
+    gamma : float, optional
+        The gamma coordinate (rotation in radians about the z axis) of the position to move to.
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_LIN_REL_TRF/MOVE_LIN_REL_TRF.py]:
```suggestion
    alpha : float
        The alpha coordinate (rotation in radians about the x axis) of the position to move to.
    beta : float
        The beta coordinate   (rotation in radians about the y axis) of the position to move to.
    gamma : float
        The gamma coordinate (rotation in radians about the z axis) of the position to move to.
```

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_LIN_REL_TRF/MOVE_LIN_REL_TRF.py [hunk 25]
```diff
@@ -0,0 +1,50 @@
...
+
+    Returns
+    -------
+    ip_address
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_LIN_REL_TRF/MOVE_LIN_REL_TRF.py]:
```suggestion
    TextBlob
```

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_POSE/MOVE_POSE.py [hunk 26]
```diff
@@ -17,11 +18,13 @@ def MOVE_POSE(
     """
     Move the robot to a specified pose in space.
 
+    Inputs
+    ------
+    ip_address
+        The IP address of the robot arm.
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_POSE/MOVE_POSE.py]: Put this under Parameters

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_POSE/MOVE_POSE.py [hunk 27]
```diff
@@ -37,7 +40,7 @@ def MOVE_POSE(
 
     Returns
     -------
-    ip_address : TextBlob
+    ip_address
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/MOVE_POSE/MOVE_POSE.py]:
```suggestion
    TextBlob
```

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/SET_BLENDING/SET_BLENDING.py [hunk 28]
```diff
@@ -0,0 +1,33 @@
...
+@safe_robot_operation
+@flojoy(deps={"mecademicpy": "1.4.0"})
+def SET_BLENDING(ip_address: TextBlob, blending: Optional[float] = 0.0) -> TextBlob:
+    """
+    The SET_BLENDING to make the moves of the robot arm smoother.
+
+    Inputs
+    ------
+    ip_address: TextBlob
+        The IP address of the robot arm.
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/SET_BLENDING/SET_BLENDING.py]: Put this under Parameters

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/SET_BLENDING/SET_BLENDING.py [hunk 29]
```diff
@@ -0,0 +1,33 @@
+from typing import Optional
+from flojoy import flojoy, TextBlob
+from PYTHON.utils.mecademic_state.mecademic_state import query_for_handle
+from PYTHON.utils.mecademic_state.mecademic_helpers import safe_robot_operation
+
+
+@safe_robot_operation
+@flojoy(deps={"mecademicpy": "1.4.0"})
+def SET_BLENDING(ip_address: TextBlob, blending: Optional[float] = 0.0) -> TextBlob:
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/SET_BLENDING/SET_BLENDING.py]:
```suggestion
def SET_BLENDING(ip_address: TextBlob, blending: float = 0.0) -> TextBlob:
```

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/SET_BLENDING/SET_BLENDING.py [hunk 30]
```diff
@@ -0,0 +1,33 @@
...
+    The SET_BLENDING to make the moves of the robot arm smoother.
+
+    Inputs
+    ------
+    ip_address: TextBlob
+        The IP address of the robot arm.
+
+    Parameters
+    ----------
+    blending: Optional[float]
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/SET_BLENDING/SET_BLENDING.py]:
```suggestion
    blending: float
```

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/SET_BLENDING/SET_BLENDING.py [hunk 31]
```diff
@@ -0,0 +1,33 @@
...
+        The blending factor to use when moving between keyframes. If not specified, the default value of 0.0 is used.
+
+    Returns
+    -------
+    ip_address
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/SET_BLENDING/SET_BLENDING.py]:
```suggestion
    TextBlob
```

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/SET_CART_LIN_VEL/SET_CART_LIN_VEL.py [hunk 32]
```diff
@@ -1,22 +1,27 @@
...
 def SET_CART_LIN_VEL(ip_address: TextBlob, v: float) -> TextBlob:
     """
     Set the robot arm's linear velocity in Cartesian coordinates.
 
-    Parameters
-    ----------
+    Inputs
+    ------
     ip_address: TextBlob
         The IP address of the robot arm.
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/SET_CART_LIN_VEL/SET_CART_LIN_VEL.py]: Put under Parameters

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/SET_JOINT_VEL/SET_JOINT_VEL.py [hunk 33]
```diff
@@ -1,26 +1,31 @@
...
-    ip_address : TextBlob
-        The IP address of the robot arm.
-
     v : float
         The angular velocity to be set for each joint.
 
+    Inputs
+    ------
+    ip_address: TextBlob
+        The IP address of the robot arm.
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/SET_JOINT_VEL/SET_JOINT_VEL.py:20]: Put under Parameters

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/SET_JOINT_VEL/SET_JOINT_VEL.py [hunk 34]
```diff
@@ -1,26 +1,31 @@
...
+
+
     Returns
     -------
-    ip_address : TextBlob
-        The IP address of the robot arm.
-    """
+    ip_address: TextBlob
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/SET_JOINT_VEL/SET_JOINT_VEL.py]:
```suggestion
    TextBlob
```

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/SET_TRF/SET_TRF.py [hunk 35]
```diff
@@ -0,0 +1,49 @@
...
+@safe_robot_operation
+@flojoy(deps={"mecademicpy": "1.4.0"})
+def SET_TRF(
+    ip_address: TextBlob,
+    rf_x: Optional[float] = 0.0,
+    rf_y: Optional[float] = 0.0,
+    rf_z: Optional[float] = 0.0,
+    rf_a: Optional[float] = 0.0,
+    rf_b: Optional[float] = 0.0,
+    rf_g: Optional[float] = 0.0,
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/SET_TRF/SET_TRF.py]:
```suggestion
    rf_x: float = 0.0,
    rf_y: float = 0.0,
    rf_z: float = 0.0,
    rf_a: float = 0.0,
    rf_b: float = 0.0,
    rf_g: float = 0.0,
```

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/SET_TRF/SET_TRF.py [hunk 36]
```diff
@@ -0,0 +1,49 @@
...
+    rf_y: Optional[float]
+        The Y coordinate of the reference plane. If not specified, the default value of 0.0 is used.
+    rf_z: Optional[float]
+        The Z coordinate of the reference plane. If not specified, the default value of 0.0 is used.
+    rf_a: Optional[float]
+        The alpha angle of the reference plane. If not specified, the default value of 0.0 is used.
+    rf_b: Optional[float]
+        The beta angle of the reference plane. If not specified, the default value of 0.0 is used.
+    rf_g: Optional[float]
+        The gamma angle of the reference plane. If not specified, the default value of 0.0 is used.
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/SET_TRF/SET_TRF.py]:
```suggestion
    rf_x: float
        The X coordinate of the reference plane. If not specified, the default value of 0.0 is used.
    rf_y: float
        The Y coordinate of the reference plane. If not specified, the default value of 0.0 is used.
    rf_z: float
        The Z coordinate of the reference plane. If not specified, the default value of 0.0 is used.
    rf_a: float
        The alpha angle of the reference plane. If not specified, the default value of 0.0 is used.
    rf_b: float
        The beta angle of the reference plane. If not specified, the default value of 0.0 is used.
    rf_g: float
        The gamma angle of the reference plane. If not specified, the default value of 0.0 is used.
```

File: blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/SET_TRF/SET_TRF.py [hunk 37]
```diff
@@ -0,0 +1,49 @@
...
+
+    Returns
+    -------
+    ip_address
+         The IP address of the robot arm.
```
Comment (39bytes) [inline blocks/HARDWARE/ROBOTICS/ARMS/MECADEMIC/SET_TRF/SET_TRF.py]:
```suggestion
    TextBlob
         The IP address of the robot arm.
```

**Review by 39bytes — COMMENTED** [review]
Almost exclusively docstring things, also a couple broken imports i believe need to be fixed

**Comment by jingxiangmo** [comment]
Resolved all PR suggestions and conversations

**Review by 39bytes — APPROVED** [review]

**Pull request merged**